tests/test_models.py

```diff
@@ -36,23 +47,210 @@ class TestAmiProductSuite:
+        with pytest.raises(ValidationError):
+            models.Version(**version)
+
+    def test_invalid_ami_id(self):
+        version = self._get_version_details()
+        version["ami_id"] = "1234567"
+        with pytest.raises(ValidationError):
+            models.Version(**version)
+
+
+class TestAmiProduct:
+    @pytest.fixture
+    def local_config(self):
+        with open("./tests/test_config.yaml", "r") as f:
+            return yaml.safe_load(f)
+
+    def test_ami_product_short_description(self, mock_boto3, local_config):
+        ami_product = models.AmiProduct(**local_config["product"])
+        assert (
+            ami_product.description.short_description == "test_short_description"
+            and ami_product.description.product_title == "test"
+        )
+
+    def test_ami_product_region_availability(self, mock_boto3, local_config):
+        ami_product = models.AmiProduct(**local_config["product"])
+        assert ami_product.region.commercial_regions == ["us-east-1", "us-east-2"]
+
+    def test_ami_product_version(self, mock_boto3, local_config):
+        ami_product = models.AmiProduct(**local_config["product"])
+        assert ami_product.version.ami_id == "ami-test"
+
+    def test_ami_product_invalid_description(self, mock_boto3, local_config):
+        local_config["product"]["description"]["categories"] = ["test"]
+        with pytest.raises(ValidationError):
+            ami_product = models.AmiProduct(**local_config["product"])
+
+    def test_ami_product_invalid_region(self, mock_boto3, local_config):
+        local_config["product"]["region"]["commercial_regions"].append("eu-west-3")
+        with pytest.raises(ValidationError):
+            ami_product = models.AmiProduct(**local_config["product"])
+
+
+class TestInstanceTypePricing:
+    @pytest.mark.parametrize(
+        "instance_type_and_pricing,expected",
+        [
+            ({"name": "c1.medium", "hourly": 0.004, "yearly": 24.528}, "0.004"),
+            ({"name": "c3.medium", "hourly": 0.012, "monthly": 57.528}, "0.012"),
+            ({"name": "c1.metal", "hourly": 0, "yearly": 0}, "0"),
+        ],
+    )
+    def test_instance_type_pricing(self, instance_type_and_pricing, expected):
+        model = models.InstanceTypePricing(**instance_type_and_pricing)
+        assert str(model.price_hourly) == expected
+
+    def test_instance_type_with_monthly_and_yearly(self):
+        instance_type_and_pricing: dict[str, Any] = {
+            "name": "c1.medium",
+            "hourly": 0.004,
+            "yearly": 24.528,
+            "monthly": 10.000,
+        }
+
+        with pytest.raises(ValidationError):
+            models.InstanceTypePricing(**instance_type_and_pricing)
+
+    def test_instance_type_without_monthly_and_yearly(self):
+        instance_type_and_pricing: dict[str, Any] = {
+            "name": "c1.medium",
+            "hourly": 0.004,
+            "yearly": 24.528,
+            "monthly": 10.000,
```

Comment:
this seems to be exercising the validation with the same inputs as the previous test. looks like we are missing some intended test cases. we should cover:

* monthly and yearly set together (already done with a recommendation above)
* annual and monthly both set to none